### PR TITLE
Add support on Azure Synapse for SAR+

### PR DESCRIPTION
**Pull request by simonzhaoms**
### Description

Azure Synapse uses slightly different mechanism for local file system.  The PR adds Synapse support for SAR+.

### Checklist:

- [X] I have followed the [contribution guidelines](../CONTRIBUTING.md) and code style for this project.
- [ ] I have added tests covering my contributions.
- [X] I have updated the documentation accordingly.
- [X] This PR is being made to `staging branch` and not to `main branch`.

### CODE REVIEW

File: contrib/sarplus/README.md [hunk 1]
```diff
@@ -102,75 +118,160 @@ spark = (
...
+
+See [Manage libraries for Apache Spark in Azure Synapse
+Analytics](https://docs.microsoft.com/en-us/azure/synapse-analytics/spark/apache-spark-azure-portal-add-libraries)
+for details on how to manage libraries in Azue Synapse.
+
+
+#### Prepare local file system for cache
+
+`pysarplus.SARPlus.recommend_k_items()` needs a local file system path
+as its second parameter for storing intermediate files during its
```
[the remark this answers is not in the document]
Comment (miguelgfierro) [reply contrib/sarplus/README.md:251]: one question about Angus' comment, the path is only used in recommen_k_items method or is it also used in the fit method?

File: contrib/sarplus/python/setup.py [hunk 2]
```diff
@@ -39,28 +37,23 @@ def __str__(self):
     classifiers=[
         "Development Status :: 5 - Production/Stable",
         "License :: OSI Approved :: MIT License",
-        "Programming Language :: Python :: 3.6",
-        "Programming Language :: Python :: 3.7",
-        "Programming Language :: Python :: 3.8",
-        "Programming Language :: Python :: 3.9",
-        "Programming Language :: Python :: 3.10",
+        "Programming Language :: Python :: 3",
```
Comment (miguelgfierro) [inline contrib/sarplus/python/setup.py:40]: We could consider keeping the specific python versions, if we only put python 3, people can think that 3.5 can be used, or future versions (3.11) that we have not test. What do you think?

One question that is not clear, python 3.6 is included or not? FYI @anargyri

**Review by miguelgfierro — APPROVED** [review]
looks great, just added minor comments

File: contrib/sarplus/python/setup.py (hunk 2 above)
Comment (anargyri) [reply contrib/sarplus/python/setup.py:40]: 3.6 is currently included. What is the reason for excluding it from SARplus?

File: contrib/sarplus/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (simonzhaoms) [reply contrib/sarplus/README.md:251]: @miguelgfierro The path is only used in `recommend_k_items()`.

@angusrtaylor `recommend_k_items()` uses C++ underneath to do the calculation which requires some file manipulations.  It is faster than another method [`recommend_k_items_slow()`](https://github.com/microsoft/recommenders/blob/137c12c6e392ccb3c604f895ca3b3af37375d5e4/contrib/sarplus/python/pysarplus/SARPlus.py#L391) that uses pure PySpark without passing the path as parameter.

File: contrib/sarplus/README.md [hunk 3]
```diff
@@ -4,6 +4,8 @@ Pronounced surplus as it's simply better if not best!
 
 [![Build Status](https://dev.azure.com/best-practices/recommenders/_apis/build/status/contrib%20sarplus?branchName=master)](https://dev.azure.com/best-practices/recommenders/_build/latest?definitionId=107&branchName=master)
```
[the remark this answers is not in the document]
Comment (simonzhaoms) [reply contrib/sarplus/README.md]: Replaced with [the status badge from GitHub Actions](https://github.com/microsoft/recommenders/blob/11b3eaa9e953615f875e6fa2a5dc61c4611740b5/contrib/sarplus/README.md?plain=1#L5)

File: contrib/sarplus/README.md [hunk 4]
```diff
@@ -102,75 +118,160 @@ spark = (
...
+
+# Install pysarplus
 pip install pysarplus
-pyspark --packages com.microsoft.sarplus:sarplus:0.5.0 --conf spark.sql.crossJoin.enabled=true
+
+# Specify sarplus maven coordinate and configure Spark environment
+pyspark --packages "${SARPLUS_MVN_COORDINATE}" \
+        --conf spark.sql.crossJoin.enabled=true \
+		--conf spark.sql.sources.default=parquet \
+		--conf spark.sql.legacy.createHiveTableByDefault=true
```
[the remark this answers is not in the document]
Comment (simonzhaoms) [reply contrib/sarplus/README.md]: Corrected

File: contrib/sarplus/README.md [hunk 5]
```diff
@@ -4,6 +4,8 @@ Pronounced surplus as it's simply better if not best!
 
 [![Build Status](https://dev.azure.com/best-practices/recommenders/_apis/build/status/contrib%20sarplus?branchName=master)](https://dev.azure.com/best-practices/recommenders/_build/latest?definitionId=107&branchName=master)
 [![PyPI version](https://badge.fury.io/py/pysarplus.svg)](https://badge.fury.io/py/pysarplus)
+[![Maven Central](https://maven-badges.herokuapp.com/maven-central/com.microsoft.sarplus/sarplus_2.12/badge.svg)](https://maven-badges.herokuapp.com/maven-central/com.microsoft.sarplus/sarplus_2.12)
+[![Maven Central](https://maven-badges.herokuapp.com/maven-central/com.microsoft.sarplus/sarplus-spark-3-2-plus_2.12/badge.svg) (Spark 3.2+)](https://maven-badges.herokuapp.com/maven-central/com.microsoft.sarplus/sarplus-spark-3-2-plus_2.12)
```
[the remark this answers is not in the document]
Comment (simonzhaoms) [reply contrib/sarplus/README.md]: Replaced with the badges from [img.shields.io](https://shields.io)

File: contrib/sarplus/python/setup.py (hunk 2 above)
Comment (simonzhaoms) [reply contrib/sarplus/python/setup.py]: @anargyri The latest [setuptools](https://pypi.org/project/setuptools/) now only supports Python 3.7+, and I also just found Python 3.6 has reached end-of-life.  Please see
* [Release Python 3.6](https://www.python.org/downloads/release/python-360/)
* [Status of Python branches](https://devguide.python.org/#status-of-python-branches)
But I reverted to Python 3.6 in alignment with PySpark now.

@miguelgfierro Agree.

Comment (anargyri) [reply contrib/sarplus/python/setup.py]: Good point about 3.6. But some people may still be using it for whatever reason, so we could still allow it.

Comment (anargyri) [inline contrib/sarplus/python/setup.py]:
```suggestion
        "Programming Language :: Python :: 3.6",
        "Programming Language :: Python :: 3.7",
        "Programming Language :: Python :: 3.8",
        "Programming Language :: Python :: 3.9",
        "Programming Language :: Python :: 3.10",
```

Comment (miguelgfierro) [reply contrib/sarplus/python/setup.py]: the commit suggestion that @angusrtaylor found now is on fashion XDD

File: contrib/sarplus/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (simonzhaoms) [reply contrib/sarplus/README.md:253]: @angusrtaylor Good idea! I'll do the refactoring in another PR.  See also the issue https://github.com/microsoft/recommenders/issues/1642

**Pull request merged**